teuthology/__init__.py

```diff
@@ -103,7 +103,9 @@ def custom_handle_error(self, context, type, value, tb):
         if not issubclass(type, Hub.SYSTEM_ERROR + Hub.NOT_ERROR):
             log.error("Uncaught exception (Hub)", exc_info=(type, value, tb))
 
-        self._origin_handle_error(context, type, value, tb)
+        # Do not call greenlet exception handler, it encounters an EIO error
```

Comment:
i think the reason why we have i/o error is that: teuthworker is running without terminal attached to it. put in other way, when a background process not properly daemonized tries to write to stdout/stderr, i/o error is expected.

we can either fix it by 

- closing its stdin,  stdout and stderr or 
- redirecting them to `/dev/null`. or, 
- just don't try to read/write to them.

because gevent not only prints the backtrace but also takes some extra measures to fix things when handling `hub.SYSTEM_ERROR`. see  https://github.com/gevent/gevent/blob/a8c1f1712bd2010cf28eb2f42774be273ce7dd4d/src/gevent/hub.py#L454 

i'd suggest put something like

```py
if context is None or issubclass(type, Hub.SYSTEM_ERROR):
  self.handle_system_error(type, value)
elif issubclass(type, Hub.NOT_ERROR):
  pass
else:
  log.error("Uncaught exception (Hub)", exc_info=(type, value, tb))
```

Reply:
Thanks, that makes sense Kefu. I've added your suggested code and also modified worker_start.sh to redirect to /dev/null. What do you think?

Reply:
Doesn't this also mean somewhere in the test there is print statement instead of log.info() inside the gevent block? If so one can remove those print and use log.info() to fix.

Reply:
The print is in the gevent code. Not feasible to fix.